### PR TITLE
fix(chat): chat input accepts DnD for blocked users

### LINKED ISSUE #16451: chat: chat input accepts DnD for blocked users
# Bug Report

## Description

When the user is blocked, chat input is disabled. However it's possible to drop images which will appear in the input, with no possibility to send or even remove. 

> [!NOTE]
> Fixing this bug is a good chance to remove `Global.dragArea` (state in global singleton)

https://github.com/user-attachments/assets/6e623b44-3224-4c94-b464-9bd559b17e7a

## Steps to reproduce
1. Go to chat with blocked user
2. Drop some image files to the app window

#### Expected behavior
The drag and drop should be just ignored as there is no active place for drop

#### Actual behavior
Chat input accepts drop and displays images

### Additional Information

- Status desktop version: `100389aa8d1d0eb3facabd9643dec216b36d721d`

### PR DESCRIPTION
**Pull request by caybro**
### What does the PR do

- do not allow DND when the StatusChatInput is disabled
- remove `Global.dragArea` variable from the Global singleton

Fixes #16451

### Affected areas

Chat/Input

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/user-attachments/assets/dbf0c944-d92f-4251-b89d-cb83fd9fda2f

### CODE REVIEW

File: ui/app/mainui/panels/DropAreaPanel.qml [hunk 1]
```diff
@@ -42,7 +39,7 @@ DropArea {
     onExited: cleanup()
 
     Loader {
-        active: root.containsDrag
+        active: root.containsDrag && root.enabled
```
Comment (caybro) [inline ui/app/mainui/panels/DropAreaPanel.qml:42]: Don't show the overlay when the chat input is disabled

File: ui/imports/shared/status/StatusChatImageLoader.qml [hunk 2]
```diff
@@ -1,8 +1,11 @@
 import QtQuick 2.15
 import QtQuick.Window 2.15
 import QtGraphicalEffects 1.15
+
 import shared.panels 1.0
 
+import utils 1.0
```
Comment (caybro) [inline ui/imports/shared/status/StatusChatImageLoader.qml:7]: Re-add import, needed for `Style` at the bottom of the file @micieslak

File: ui/imports/utils/Global.qml [hunk 3]
```diff
@@ -76,6 +75,9 @@ QtObject {
     signal openLinkWithConfirmation(string link, string domain)
     signal activateDeepLink(string link)
 
+    // Global drop area
+    signal droppedOnValidScreen(var drop)
```
Comment (caybro) [inline ui/imports/utils/Global.qml:79]: Here we just relay the global DND info via a signal (stateless)

File: ui/app/mainui/AppMain.qml [hunk 4]
```diff
@@ -1356,7 +1356,8 @@ Item {
                                 Binding {
                                     target: rootDropAreaPanel
                                     property: "enabled"
-                                    value: chatLayoutContainer.currentIndex === 0 // Meaning: Chats / channels view
+                                    value: chatLayoutContainer.currentIndex === 0 && // Meaning: Chats / channels view
```
Comment (caybro) [inline ui/app/mainui/AppMain.qml:1359]: Disable the overlay/DND when the chat input is disabled

**Review by jrainville — APPROVED** [review]

File: ui/app/mainui/AppMain.qml [hunk 5]
```diff
@@ -2135,6 +2137,7 @@ Item {
 
         width: appMain.width
         height: appMain.height
+        onDroppedOnValidScreen: (drop) => Global.droppedOnValidScreen(drop)
```
Comment (micieslak) [inline ui/app/mainui/AppMain.qml:2140]: I think that `DropAreaPanel` could and should be moved into `StatusChatInpput`. Then there is no need to use `Global` as a proxy, status chat input is self-contained regarding dnd support and no bindin to `chatInputEnabled` is needed then. I would give a try, if I'm not missing anything, it should simplify the code.

Comment (caybro) [reply ui/app/mainui/AppMain.qml]: Fixed; much cleaner now

Comment (micieslak) [reply ui/app/mainui/AppMain.qml]: So nice! Now that functionality is in just 1 place instead of ~10 places :D

**Review by micieslak — APPROVED** [review]
Cool!

**Pull request merged**